x/ecocredit/server/marketplace/add_ask_denom.go

```diff
@@ -0,0 +1,33 @@
+package marketplace
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+// AddAskDenom is a gov handler method that adds a denom to the list of approved denoms that may be used in the
+// marketplace.
+func (k Keeper) AddAskDenom(ctx sdk.Context, p *marketplace.AllowedDenomProposal) error {
+	if p == nil {
+		return sdkerrors.ErrInvalidRequest.Wrap("nil proposal")
+	}
+	if err := p.ValidateBasic(); err != nil {
+		return err
+	}
+	askDenom := p.AllowedDenom
+	if err := k.stateStore.AllowedDenomTable().Insert(sdk.WrapSDKContext(ctx), &api.AllowedDenom{
```

Comment:
Do we need remove ask denom? With params, we could update `KeyAllowedAskDenoms`  param to remove denom.